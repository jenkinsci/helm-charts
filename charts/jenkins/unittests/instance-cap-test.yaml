suite: Instance cap tests
release:
  name: my-release
  namespace: my-namespace
templates:
  - jcasc-config.yaml
tests:
  - it: default-cap
    asserts:
      - isKind:
          of: ConfigMap
      - hasDocuments:
          count: 1
      - isNotEmpty:
          path: data["jcasc-default-config.yaml"]
      - matchRegex:
          path: metadata.labels["helm.sh/chart"]
          pattern: ^jenkins-
      - matchSnapshot:
          path: data["jcasc-default-config.yaml"]
      - equal:
          path: metadata.namespace
          value: my-namespace

  - it: limited-cap
    set:
      agent.instanceCap: 5
    asserts:
      - isKind:
          of: ConfigMap
      - hasDocuments:
          count: 1
      - isNotEmpty:
          path: data["jcasc-default-config.yaml"]
      - matchRegex:
          path: metadata.labels["helm.sh/chart"]
          pattern: ^jenkins-
      - matchSnapshot:
          path: data["jcasc-default-config.yaml"]
      - equal:
          path: metadata.namespace
          value: my-namespace
