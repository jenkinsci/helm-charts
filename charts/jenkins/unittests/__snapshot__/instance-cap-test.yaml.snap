default-cap:
  1: |
    |-
      jenkins:
        authorizationStrategy:
          loggedInUsersCanDoAnything:
            allowAnonymousRead: false
        securityRealm:
          local:
            allowsSignup: false
            enableCaptcha: false
            users:
            - id: "${chart-admin-username}"
              name: "Jenkins Admin"
              password: "${chart-admin-password}"
        disableRememberMe: false
        mode: NORMAL
        numExecutors: 0
        labelString: ""
        projectNamingStrategy: "standard"
        markupFormatter:
          plainText
        clouds:
        - kubernetes:
            containerCapStr: "10"
            defaultsProviderTemplate: ""
            connectTimeout: "5"
            readTimeout: "15"
            jenkinsUrl: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080"
            jenkinsTunnel: "my-release-jenkins-agent.my-namespace.svc.cluster.local:50000"
            skipTlsVerify: false
            usageRestricted: false
            maxRequestsPerHostStr: "32"
            retentionTimeout: "5"
            waitForPodSec: "600"
            name: "kubernetes"
            namespace: "my-namespace"
            restrictedPssSecurityContext: false
            serverUrl: "https://kubernetes.default"
            credentialsId: ""
            podLabels:
            - key: "jenkins/my-release-jenkins-agent"
              value: "true"
            templates:
              - name: "default"
                namespace: "my-namespace"
                id: 022135f78097870740d147ab5422584a59f31718160dcb9f0dd9c29da98174d2
                containers:
                - name: "jnlp"
                  alwaysPullImage: false
                  args: "^${computer.jnlpmac} ^${computer.name}"
                  envVars:
                    - envVar:
                        key: "JENKINS_URL"
                        value: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080/"
                  image: "jenkins/inbound-agent:3307.v632ed11b_3a_c7-2"
                  privileged: "false"
                  resourceLimitCpu: 512m
                  resourceLimitMemory: 512Mi
                  resourceRequestCpu: 512m
                  resourceRequestMemory: 512Mi
                  ttyEnabled: false
                  workingDir: /home/jenkins/agent
                idleMinutes: 0
                instanceCap: 2147483647
                label: "my-release-jenkins-agent "
                nodeUsageMode: "NORMAL"
                podRetention: Never
                showRawYaml: true
                serviceAccount: "default"
                slaveConnectTimeoutStr: "100"
                yamlMergeStrategy: override
                inheritYamlMergeStrategy: false
        slaveAgentPort: 50000
        crumbIssuer:
          standard:
            excludeClientIPFromCrumb: true
      security:
        apiToken:
          creationOfLegacyTokenEnabled: false
          tokenGenerationOnCreationEnabled: false
          usageStatisticsEnabled: true
      unclassified:
        location:
          url: http://my-release-jenkins:8080
limited-cap:
  1: |
    |-
      jenkins:
        authorizationStrategy:
          loggedInUsersCanDoAnything:
            allowAnonymousRead: false
        securityRealm:
          local:
            allowsSignup: false
            enableCaptcha: false
            users:
            - id: "${chart-admin-username}"
              name: "Jenkins Admin"
              password: "${chart-admin-password}"
        disableRememberMe: false
        mode: NORMAL
        numExecutors: 0
        labelString: ""
        projectNamingStrategy: "standard"
        markupFormatter:
          plainText
        clouds:
        - kubernetes:
            containerCapStr: "10"
            defaultsProviderTemplate: ""
            connectTimeout: "5"
            readTimeout: "15"
            jenkinsUrl: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080"
            jenkinsTunnel: "my-release-jenkins-agent.my-namespace.svc.cluster.local:50000"
            skipTlsVerify: false
            usageRestricted: false
            maxRequestsPerHostStr: "32"
            retentionTimeout: "5"
            waitForPodSec: "600"
            name: "kubernetes"
            namespace: "my-namespace"
            restrictedPssSecurityContext: false
            serverUrl: "https://kubernetes.default"
            credentialsId: ""
            podLabels:
            - key: "jenkins/my-release-jenkins-agent"
              value: "true"
            templates:
              - name: "default"
                namespace: "my-namespace"
                id: cd8fb292ee436120e71a9ecdb86afa241ee461722fc188796fe0b5409810db14
                containers:
                - name: "jnlp"
                  alwaysPullImage: false
                  args: "^${computer.jnlpmac} ^${computer.name}"
                  envVars:
                    - envVar:
                        key: "JENKINS_URL"
                        value: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080/"
                  image: "jenkins/inbound-agent:3307.v632ed11b_3a_c7-2"
                  privileged: "false"
                  resourceLimitCpu: 512m
                  resourceLimitMemory: 512Mi
                  resourceRequestCpu: 512m
                  resourceRequestMemory: 512Mi
                  ttyEnabled: false
                  workingDir: /home/jenkins/agent
                idleMinutes: 0
                instanceCap: 5
                label: "my-release-jenkins-agent "
                nodeUsageMode: "NORMAL"
                podRetention: Never
                showRawYaml: true
                serviceAccount: "default"
                slaveConnectTimeoutStr: "100"
                yamlMergeStrategy: override
                inheritYamlMergeStrategy: false
        slaveAgentPort: 50000
        crumbIssuer:
          standard:
            excludeClientIPFromCrumb: true
      security:
        apiToken:
          creationOfLegacyTokenEnabled: false
          tokenGenerationOnCreationEnabled: false
          usageStatisticsEnabled: true
      unclassified:
        location:
          url: http://my-release-jenkins:8080
